### PR TITLE
Issue 16: Modify sa_svp_buffer_copy and sa_svp_buffer_write to support multiple offsets

### PR DESCRIPTION
**Pull request by ericeberry**

### CODE REVIEW

File: reference/src/taimpl/src/ta_sa_svp_buffer_copy_blocks.c [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+            ERROR("svp_store_acquire_exclusive failed");
+            break;
+        }
+
+        svp_buffer_t* out_svp_buffer = svp_get_buffer(out_svp);
+        svp_buffer_t* in_svp_buffer = svp_get_buffer(in_svp);
+        for (size_t i = 0; i < blocks_length; i++) {
+            if (!svp_copy(out_svp_buffer, blocks[i].out_offset, in_svp_buffer, blocks[i].in_offset, blocks[i].length)) {
+                ERROR("svp_copy failed");
+                status = SA_STATUS_BAD_SVP_BUFFER;
```
Comment (mhabrat) [inline reference/src/taimpl/src/ta_sa_svp_buffer_copy_blocks.c:72]: Recommend avoiding general terms like good or bad in naming and use terms like invalid.

File: reference/src/clientimpl/src/sa_svp_buffer_copy_blocks.c [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+        svp_buffer_copy_blocks->api_version = API_VERSION;
+        svp_buffer_copy_blocks->out_svp_buffer = out;
+        svp_buffer_copy_blocks->in_svp_buffer = in;
+        svp_buffer_copy_blocks->blocks_length = blocks_length;
+
+        CREATE_PARAM(param1, blocks, sizeof(sa_svp_block) * blocks_length);
+        param1_size = sizeof(sa_svp_block) * blocks_length;
+        param1_type = TA_PARAM_IN;
+                                                     // clang-format off
```
[the remark this answers is not in the document]
Comment (ericeberry) [reply reference/src/clientimpl/src/sa_svp_buffer_copy_blocks.c]: Fixed

File: reference/src/taimpl/src/ta_sa_svp_buffer_copy_blocks.c (hunk 1 above)
Comment (ericeberry) [reply reference/src/taimpl/src/ta_sa_svp_buffer_copy_blocks.c]: Done. This was a really big change by the way.

File: reference/src/client/include/sa_svp.h [hunk 3]
```diff
@@ -199,6 +199,30 @@ sa_status sa_svp_buffer_copy(
...
+ * @return Operation status. Possible values are:
+ * + SA_STATUS_OK - Operation succeeded.
+ * + SA_STATUS_NULL_PARAMETER - out, out_offset or in is NULL.
+ * + SA_STATUS_BAD_PARAMETER - Reading or writing past the end of the SVP buffer detected.
+ * + SA_STATUS_BAD_SVP_BUFFER - SVP buffer is not fully contained withing SVP memory region.
+ * + SA_STATUS_OPERATION_NOT_SUPPORTED - Implementation does not support the specified operation.
+ * + SA_STATUS_SELF_TEST - Implementation self-test has failed.
+ * + SA_STATUS_INTERNAL_ERROR - An unexpected error has occurred.
+ */
+sa_status sa_svp_buffer_copy_blocks(
```
[the remark this answers is not in the document]
Comment (ericeberry) [reply reference/src/client/include/sa_svp.h]: Consolidated two APIs

**Review by mhabrat — APPROVED** [review]

**Pull request merged**